### ISSUE TITLE
Start script from turn --

### ISSUE THREAD

**Issue by Marrionetta** [Fate-Grand-Automata/FGA#1019]
### Preparation

- [X] No one has asked for this feature before in other issues (even the closed ones)

### Describe the feature

I tried looking thru the first couple of pages to see if someone thought of this but there were just so many to look thru so...the checkmark is half false. 

So anyway...I do a ton of testing with optimal builds to see if I can make the most efficient set ups possible but either I make a lot of mistakes or fail to consider things that would break the script. Usually I'll do a manual run first but sometimes I I'll push the wrong buttons on the Script UI so when the script breaks, I have to do the rest of the fight manually and test the changes in another run.

Ideally, it would be super convenient, not to mention AP cost effective if we could choose from which point to start the script. For example,

If turn one completes with no problems, but breaker turn 2, you'd stop the script and reset the game to prevent the turn from completing the botched commands.

Once you get back in, you're still in turn two making the script unusable until you either abandon or complete the run.

I'm proposing a setting you can enable that prompts you to choose which turn the script begins from. Ideally it would consider each NP# to be the current turn's final action. So from left to right, it would read it as such

A,B,C,NP# (turn 1 end), D,E,F,NP# (turn 2 end) and so on and would start the turn using the first action after the end.

So if I chose start with turn 2, it would take the first action after the first NP# or turn 1 end flag and start with action D.

If you choose start as normal or start from turn 1, it would simply dlstart from the beginning, and so on.

This would allow people to continue testing the full script without needing to expend as much ap as you would have to otherwise.

**Comment by reconman** [comment]
The problem with your motivation is that triggering a skill too early cannot be reverted, so most faulty runs will not be automatically completable anyways.

**Comment by Marrionetta** [comment]
No. I think you misunderstand. What I mean is, if for whatever reason, your battle phase doesn't go as planned, you can reset the game and and redo the turn. But if you're on turn two, you cannot restart the script as it will inherently do everything wrong.

Like for example, I go into turn two and, it uses the skills but during the battle phase, it selects the wrong N p card because of my error. I can reset it, correct the error, finish the turn and launch from turn 3 to continue testing instead of having to wait for a brand new mission.

It may seem minute but 40 ap is an expensive cost and if you for whatever reason are making a ton of mistakes it adds up big time. I feel like being able to start from a certain turn would be useful for cutting down on a lot of ap.

Sometimes for reasons you can't figure out, the script won't even bother activating skills and will just go into battle. Sometimes it will ignore the NO PROBLEM.d select three cards.

In those situations, being able to resume the script from the following turn could save time, weed out further potential errors and save ap.

My only other alternative at this point is to make three seperate scripts per load out to use depending on what turn I need to run from which is pretty overkill

**Comment by Marrionetta** [comment]
Another example would be; I was testing out a jarcher arts loop with Tama/Jarch/Paracel.

On the manual test, Jarcher was able to get a maximum refund off her NPcard alone without using her S2 or a Mystic code skill but on the auto test, she came up 10% short. So I had to stop the script, reset and have my MC give a remaining 10% charge to her. I'd complete the turn manually and start off turn 3. 

At that point, I'd get to see if her NPcard fully refunded her NPgague or not. If it did, I could run the script from turn 3. If not, I'd set it to use her S2 THEN use her NPcard.

Normally testing all this would cost me 120AP but with the feature I'm proposing, it would only cost 40.

**Comment by Marrionetta** [comment]
> The problem with your motivation is that triggering a skill too early cannot be reverted, so most faulty runs will not be automatically completable anyways.

I edited my initial post to add greater motivation for this feature.

**Comment by reconman** [comment]
> Basically FGA on a near ridiculous frequency, at least for me, will consistently function perfectly turn one, then completely die and ignore all instructions for turn 2, going straight into battle phase.

Sounds like an issue with the stage counter detection. You can try Thresholded Stage Counter Detection in More Options.

If that doesn't help, increase the min stage similarity in Fine Tune.

**Comment by Marrionetta** [comment]
> > Basically FGA on a near ridiculous frequency, at least for me, will consistently function perfectly turn one, then completely die and ignore all instructions for turn 2, going straight into battle phase.
> 
> Sounds like an issue with the stage counter detection. You can try Thresholded Stage Counter Detection in More Options.
> 
> If that doesn't help, increase the min stage similarity in Fine Tune.

Oh. Okie. Though there are two settings that both include part of what you said so are you saying to increase both
"Min Similarity" and "Stage Counter Similarity"? Or just the stage counter one?

**Comment by reconman** [comment]
Thresholded is a different way of detecting the current stage.

Stage Counter Similarity controls how similar the stage counters have to be to be detected as the same number.

Both have effects, but I would try one first and then the other if it doesn't help.

**Comment by Marrionetta** [comment]
> Thresholded is a different way of detecting the current stage.
> 
> Stage Counter Similarity controls how similar the stage counters have to be to be detected as the same number.
> 
> Both have effects, but I would try one first and then the other if it doesn't help.

Huh! Changing the Min to 90% got it working again ouo

Is there any reason not to have it at 100%? XD

**Comment by reconman** [comment]
If you set it too high, turns aren't detected anymore, so it will trigger skills from the 2nd wave while you're still facing against the first wave.

**Comment by Marrionetta** [comment]
Ah. Yeah that would be a problem. Well hopefully 90% min is the sweet spot. I've edited my main post using my arts loop example instead for the motivation. Thanks for the stage counter help~
